crates/optimism/trie/src/provider.rs

```diff
@@ -207,3 +207,26 @@ impl<'a, Storage: OpProofsStore> BytecodeReader for OpProofsStateProviderRef<'a,
         self.latest.bytecode_by_hash(code_hash)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::InMemoryProofsStorage;
+    use reth_provider::noop::NoopProvider;
+
+    #[test]
+    #[cfg(not(feature = "metrics"))]
```

Comment:
```suggestion
#[cfg(all(test, not(feature = "metrics")))]
mod tests {
    use super::*;
    use crate::InMemoryProofsStorage;
    use reth_provider::noop::NoopProvider;

    #[test]
```
this should get CI passing